crates/chain/src/tx_graph.rs

```diff
@@ -52,28 +54,82 @@
 //! let additions = graph.apply_update(update);
 //! assert!(additions.is_empty());
 //! ```
-use crate::{collections::*, ForEachTxOut};
+
+use crate::{collections::*, BlockAnchor, BlockId, ChainOracle, ForEachTxOut, ObservedIn};
 use alloc::vec::Vec;
 use bitcoin::{OutPoint, Transaction, TxOut, Txid};
-use core::ops::RangeInclusive;
+use core::{
+    convert::Infallible,
+    ops::{Deref, RangeInclusive},
+};
 
 /// A graph of transactions and spends.
 ///
 /// See the [module-level documentation] for more.
 ///
 /// [module-level documentation]: crate::tx_graph
-#[derive(Clone, Debug, PartialEq, Default)]
-pub struct TxGraph {
-    txs: HashMap<Txid, TxNode>,
+#[derive(Clone, Debug, PartialEq)]
+pub struct TxGraph<A = BlockId> {
```

Comment:
The default should be `A = ()`. The existing API should just use `TxGraph` and nothing should change there.

Reply:
Done in 8c906170c96919cd8c65e306b8351fe01e139fd4